### ISSUE TITLE
Configuration file seems to be completely ignored

### ISSUE THREAD

**Issue by martinlicht** [markdownlint/markdownlint#519]
## Description
Configuration file seems to be ignored. There is no effect on the program when rules are excluded.

### Environment

> $ lsb_release -a
> No LSB modules are available.
> Distributor ID: Ubuntu
> Description:    Ubuntu 24.04.2 LTS
> Release:        24.04
> Codename:       noble
> $ ruby --version 
> ruby 3.2.3 (2024-01-18 revision 52bb2ac0a6) [x86_64-linux-gnu]
> $ mdl --version
> 0.13.0
> **MDL Version 0.13.0-4 installed on Ubuntu via apt** 

### Expected Behavior
The configuration file should be read and used by the program 

### Actual Behavior
The configuration seems to have no effect.

## Replication Case
The following shell commands replicate the problem:

```
echo " - - Test   " > deleteme.md
echo "exclude_rule 'MD006'" > .mdlrc
cat deleteme.md
cat .mdlrc
mdl --verbose --config .mdlrc deleteme.md
mdl --verbose -c .mdlrc deleteme.md
mdl --verbose deleteme.md
mdl --verbose --config .mdlrc -r ~MD006 deleteme.md
mdl --verbose -c .mdlrc -r ~MD006 deleteme.md
mdl --verbose -r ~MD006 deleteme.md
```

The rules violated are MD006 and MD009. I want to explicitly disable MD006 but this is completely ignored in the first and two cases of `mdl`, despite the configuration file. The third `mdl` checks everything, as expected. The last three examples ignore rule MD006, as expected. This seems to solely rely on the command line option. 

I have read the documentation but maybe I misunderstand how to use the program. If the above example works with you, please let me know configuration details that might matter.

**Comment by martinlicht** [comment]
My bad, I have realized this is a duplicate of #456

**Comment by nbehrnd** [comment]
@martinlicht My recommendation is to address the task with a style file.

Create a file for instance of name `mystyle.rb` with the content of

```mystyle.rb
all
exclude_rule 'MD006'
```

Then compare the report by mdl by either default

```bash
$ mdl ./deleteme.md 
./deleteme.md:1: MD006 Consider starting bulleted lists at the beginning of the line
./deleteme.md:1: MD009 Trailing spaces
```

with

```bash
$ mdl -s ./mystyle.rb deleteme.md 
deleteme.md:1: MD009 Trailing spaces
deleteme.md:1: MD041 First line in file should be a top level header
```

where  rule `MD006` now no longer is considered.  Back in the day when preparing the package for Debian (around https://github.com/markdownlint/markdownlint/issues/310), I presumed the man page' examples equally would become available for Ubuntu.  (Though the style's file may be of any name, and of any file extension.)

**Comment by martinlicht** [comment]
Thank you. I guess if one is used to clang-tidy config files, which contain command line parameters and style conventions in one single file, then the mdl terminology might confuse at first.
